### PR TITLE
[PRW-2.0] (part 2) Removed automatic negotiation, updates for the latest spec semantics in remote pkg

### PR DESCRIPTION
**Pull request by bwplotka**
This is chained PR for review readability. See previous PR (chain1) first: https://github.com/prometheus/prometheus/pull/14335

Spec: https://prometheus.io/docs/specs/remote_write_spec_2_0

Supersedes https://github.com/prometheus/prometheus/pull/13968

## High-lvl changes

* Config (https://github.com/prometheus/prometheus/pull/14335)
* Removal of negotiation pieces introduced in https://github.com/prometheus/prometheus/pull/13921 as we decided to skip auto-neg in 2.0 (head handler, probing, automatic upg/downgr)
* Moved useful symbolization mechanisms next to proto writev2 pkg
* Enhanced/refactored some tests

### CODE REVIEW

File: storage/remote/queue_manager.go [hunk 1]
```diff
@@ -1638,20 +1591,9 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue *queue) {
...
-				if sendErr == nil || !errors.As(sendErr, &pErr) {
-					// No error, or error wasn't a 406 or 400, so we can stop trying.
-					break
-				}
-				// If we get either of the two errors (406, 400) bundled in ErrRenegotiate we loop and re-negotiate.
-				// TODO(alexg) - add retry/renegotiate metrics here
-			}
 
+			sendBatch(batch, s.qm.protoMsg, s.qm.enc, false)
+			// TODO(bwplotka): Previously the return was between popular and send, double check.
```
Comment (npazosmendez) [inline storage/remote/queue_manager.go:1596]: What's this? Is it important?

File: storage/remote/queue_manager_test.go [hunk 2]
```diff
@@ -65,89 +66,80 @@ func newHighestTimestampMetric() *maxTimestamp {
...
-	writeConfig := baseRemoteWriteConfig("http://test-storage.com")
-	writeConfig.QueueConfig = queueConfig
-
-	conf := &config.Config{
-		GlobalConfig: config.DefaultGlobalConfig,
-		RemoteWriteConfigs: []*config.RemoteWriteConfig{
-			writeConfig,
+		{
+			success: false, name: "v1 talks to (broken) v2 that tries to unmarshal v1 payload with v2 proto", senderProtoMsg: config.RemoteWriteProtoMsgV1, receiverProtoMsg: config.RemoteWriteProtoMsgV2, steps: []contentNegotiationStep{
+				{behaviour: nil, attemptString: "0,ok"}, // It's ok, because payload can be unmarshaled just fine, but it's empty.
```
Comment (npazosmendez) [inline storage/remote/queue_manager_test.go:139]: ACAIFT `attemptString` is not used anywhere. After these changes, only the `behaviour` fields from the `steps: []contentNegotiationStep` are used, and only tell the TestClient to return a sequence of errors. I would replace the `steps : []contentNegotiationStep` with `errsSeq: []error` or similar

**Review by npazosmendez — APPROVED** [review]
LGTM - Two minor non-blocking comments/questions

File: storage/remote/queue_manager_test.go (hunk 2 above)
Comment (bwplotka) [reply storage/remote/queue_manager_test.go:139]: hah lol, you are right. Thanks! will fix

File: storage/remote/write_handler.go [hunk 3]
```diff
@@ -174,44 +140,34 @@ func (h *writeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-	case RemoteWriteVersion20HeaderValue:
-		// 2.0 request.
-		var reqMinStr writev2.Request
-		if err := proto.Unmarshal(decompressed, &reqMinStr); err != nil {
-			level.Error(h.logger).Log("msg", "Error decoding remote write request", "err", err.Error())
+	case config.RemoteWriteProtoMsgV2:
+		var req writev2.Request
+		if err := proto.Unmarshal(decompressed, &req); err != nil {
+			// TODO(bwplotka): Add more context to responded error?
+			level.Error(h.logger).Log("msg", "Error decoding remote write request", "protobuf_message", msg, "err", err.Error())
```
Comment (cstyan) [inline storage/remote/write_handler.go:166]: we could at least include the proto message type, or make it `error decoding remote write request` and `error decoding v2 remote write request`?

**Review by cstyan — APPROVED** [review]
just one minor comment

File: storage/remote/write_handler.go (hunk 3 above)
Comment (bwplotka) [reply storage/remote/write_handler.go:166]: I added, but this question in TODO is more towards HTTP response. Will leave the TODO for now.

**Pull request merged**